src/communication/tlsvchannel.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mbedtls/debug.h>
+#include <psa/crypto.h>
+
+#include <aos/common/crypto/mbedtls/driverwrapper.hpp>
+#include <aos/common/cryptoutils.hpp>
+#include <aos/iam/certhandler.hpp>
+
+#include "tlsvchannel.hpp"
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+TLSChannel::~TLSChannel()
+{
+    mbedtls_x509_crt_free(&mCertChain);
+    mbedtls_x509_crt_free(&mCACert);
+    mbedtls_pk_free(&mPrivKeyCtx);
+    mbedtls_entropy_free(&mEntropy);
+    mbedtls_ctr_drbg_free(&mCtrDrbg);
+    mbedtls_ssl_config_free(&mConf);
+    mbedtls_ssl_free(&mSsl);
+
+    if (mKeyID != 0) {
+        AosPsaRemoveKey(mKeyID);
+    }
+}
+
+aos::Error TLSChannel::Init(const aos::String& xsReadPath, const aos::String xsWritePath, const aos::String& certType,
```

Comment:
* xsReadPath and xsWritePath not used;
* vChannel -> channel and make it as first parameter.

Reply:
Done